contribute/design/1.x/apis/cstor_volume.md

```diff
@@ -0,0 +1,316 @@
+	CVStatuses []CVStatus `json:"volumeStatus"`
+}
+
+// CVStatus stores the status of a CstorVolume obtained from response
+type CVStatus struct {
+	Name            string          `json:"name"`
+	Status          string          `json:"status"`
+	ReplicaStatuses []ReplicaStatus `json:"replicaStatus"`
+}
+
+// ReplicaStatus stores the status of replicas
+type ReplicaStatus struct {
+	ID                string `json:"replicaId"`
+	Mode              string `json:"mode"`
+	CheckpointedIOSeq string `json:"checkpointedIOSeq"`
+	InflightRead      string `json:"inflightRead"`
+	InflightWrite     string `json:"inflightWrite"`
+	InflightSync      string `json:"inflightSync"`
+	UpTime            int    `json:"upTime"`
+	Quorum            string `json:"quorum"`
+}
+
+// CStorVolumeCondition contains details about state of cstorvolume
+type CStorVolumeCondition struct {
+	Type   CStorVolumeConditionType `json:"type" protobuf:"bytes,1,opt,name=type,casttype=CStorVolumeConditionType"`
+	Status ConditionStatus          `json:"status" protobuf:"bytes,2,opt,name=status,casttype=ConditionStatus"`
+	// Last time we probed the condition.
+	// +optional
+	LastProbeTime metav1.Time `json:"lastProbeTime,omitempty" protobuf:"bytes,3,opt,name=lastProbeTime"`
+	// Last time the condition transitioned from one status to another.
+	// +optional
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty" protobuf:"bytes,4,opt,name=lastTransitionTime"`
+	// Unique, this should be a short, machine understandable string that gives the reason
+	// for condition's last transition. If it reports "ResizePending" that means the underlying
+	// cstorvolume is being resized.
+	// +optional
+	Reason string `json:"reason,omitempty" protobuf:"bytes,5,opt,name=reason"`
+	// Human-readable message indicating details about last transition.
+	// +optional
+	Message string `json:"message,omitempty" protobuf:"bytes,6,opt,name=message"`
+}
+
+// CStorVolumeConditionType is a valid value of CStorVolumeCondition.Type
+type CStorVolumeConditionType string
+
+const (
+	// CStorVolumeResizing - a user trigger resize of pvc has been started
+	CStorVolumeResizing CStorVolumeConditionType = "Resizing"
+)
+
+// ConditionStatus states in which state condition is present
+type ConditionStatus string
+
+// These are valid condition statuses. "ConditionInProgress" means corresponding
+// condition is inprogress. "ConditionSuccess" means corresponding condition is success
+const (
+	// ConditionInProgress states resize of underlying volumes are in progress
+	ConditionInProgress ConditionStatus = "InProgress"
+	// ConditionSuccess states resizing underlying volumes are successfull
+	ConditionSuccess ConditionStatus = "Success"
+)
+```
+# Changes to existing types
+
+- Remvoe the `Status` field of the `CStorvolumeSpec`. This should not be too problematic but `Status` filed has been used to   set the initial `init` state of `CStorVolume` to be used in other SAAS application like openebs-director etc.
```

Comment:
there are lot of spelling mistakes in this PR.. in this line itself, there are two of them.. Please verify for all spelling mistakes.
And, which field has been removed? I don't understand

Reply:
I still see 'CStorVolumeStatus' being there in the CR

Reply:
done